src/Api/Domain.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Nexylan packages.
+ *
+ * (c) Nexylan SAS <contact@nexylan.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Nexy\Gandi\Api;
+
+/**
+ * @author Jérôme Pogeant <p-jerome@hotmail.fr>
+ */
+class Domain extends AbstractApi
+{
+    /**
+     * @param array $domain
+     * @param array $options
+     *
+     * @return array
+     */
+    public function isAvailable(array $domain, array $options = null): array
+    {
+        $results = $this->gandi->setup()->domain->available($domain, $options);
+
+        foreach ($results as $domain => $result) {
+            if ('pending' === $result) {
+                usleep(700000);
```

Comment:
WTF?

Reply:
@Soullivaneuh This is like the Gandi API doc : http://doc.rpc.gandi.net/domain/usage.html#tab-php-20

Reply:
This belongs to the application code.

Here, you just have to return the result, pending or not.